### PR TITLE
CMS-435: Add new org:update-projects-info command.

### PR DESCRIPTION
**Pull request by kporras07**
### Overview
This pull request adds a new command that takes a csv as argument and goes through all of the projects and update project info for each line.

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes 
| Has tests?    | no
| BC breaks?    | no     
| Deprecations? | no 

### Description
- Also accept badge label in the project:update-info command argument
- Adds org:update-projects-info command
- Refactor project:info-update into trait to make it reusable

### Notes
- I ran this command like this `./update-tool org:update-projects-info projects.csv --update-codeowners --update-support-level-badge` against this csv file: https://gist.github.com/kporras07/5daa2c67ba6ca8f808120a8e6234861c and it created this PR: https://github.com/kporras07/wordpress-network/pull/23/files

### CODE REVIEW

**Comment by kporras07** [comment]
This depends on https://github.com/g1a/hubph/pull/46

Once merged and tagged hubph release I'll update this PR but anything else could be reviewed now

File: src/Cli/OrgCommands.php
Comment (greg-1-anderson) [inline src/Cli/OrgCommands.php:170]: Instead of skipping the header row, it would be better to call a search function that returned `[3, 5, 23, 97]` based on the index that each header row label is found at.

File: src/Cli/OrgCommands.php
Comment (greg-1-anderson) [inline src/Cli/OrgCommands.php:190]: $ownerSource is just something like `file`, it won't be set if $codeowners is empty

File: src/Cli/OrgCommands.php
Comment (kporras07) [reply src/Cli/OrgCommands.php:170]: you mean using the header row to lookup for the right column number based on column titles?

File: src/Cli/OrgCommands.php
Comment (kporras07) [reply src/Cli/OrgCommands.php:190]: It looks like it could be "file", "api" or "guess". Then, my question is, should it be the same result no matter the source?

File: src/Cli/OrgCommands.php
Comment (greg-1-anderson) [reply src/Cli/OrgCommands.php:170]: Yep. Then the script does not become fragile if the composition of the spreadsheet changes over time (as long as the titles of the columns we depend on remain constant)

File: src/Cli/OrgCommands.php
Comment (greg-1-anderson) [reply src/Cli/OrgCommands.php:190]: Probably for now. Having a guess / don't guess option might be handy. Optional.

File: src/Util/ProjectUpdateTrait.php
Comment (greg-1-anderson) [inline src/Util/ProjectUpdateTrait.php:15]: This seems a little heavyweight for a trait, especially with the logger injection. Instead, make this a utility class, and inject the logger via the constructor.

Comment (greg-1-anderson) [reply src/Util/ProjectUpdateTrait.php:15]: Or use setLogger like workingCopy does.

File: src/Cli/OrgCommands.php
Comment (kporras07) [reply src/Cli/OrgCommands.php:190]: If it's "file" then it's already there and updating it doesn't make sense. So, I guess we're down to updating it only if "api" or "guess". Maybe adding --only-api and --only-guess options?

File: src/Util/ProjectUpdateTrait.php
Comment (greg-1-anderson) [inline src/Util/ProjectUpdateTrait.php:81]: Is this idempotent? It sort of looks like you'll end up with two support level badges if you run this on a README that already has a support level badge.

Comment (greg-1-anderson) [reply src/Util/ProjectUpdateTrait.php:81]: (At a different support level, that is)

Comment (kporras07) [reply src/Util/ProjectUpdateTrait.php:81]: That wouldn't run because of SupportLevel::compareSupportLevelFromReadmeAndBadge

**Review by greg-1-anderson — CHANGES_REQUESTED** [review]
Looking pretty good; check idempotency, made a couple of code change requests.

File: src/Util/ProjectUpdateTrait.php
Comment (kporras07) [reply src/Util/ProjectUpdateTrait.php:81]: ohhh, different support level. What should be the outcome with different levels?

File: src/Cli/OrgCommands.php
Comment (kporras07) [reply src/Cli/OrgCommands.php]: Implemented new codeowners-only-api and codeowners-only-guess options.

File: src/Util/ProjectUpdateTrait.php
Comment (greg-1-anderson) [reply src/Util/ProjectUpdateTrait.php]: Replace the existing support level badge with a new one.

Comment (kporras07) [reply src/Util/ProjectUpdateTrait.php]: Fixed

File: src/Cli/OrgCommands.php
Comment (greg-1-anderson) [inline src/Cli/OrgCommands.php:185]: It would be useful to be able to specify the title and/or the JIRA story ("[CMS-1234]"), so that folks will know where to go for more info when a PR is opened against their project.

Comment (kporras07) [reply src/Cli/OrgCommands.php]: Definitely. Done. Added as an option and I kept this text as the default option.

**Review by greg-1-anderson — APPROVED** [review]

**Pull request merged**
